File: Gnuify/src/main/java/com/aliucord/plugins/Gnuify.java
package com.aliucord.plugins;

import android.content.Context;

import androidx.annotation.NonNull;

import com.aliucord.entities.Plugin;
import com.aliucord.api.CommandsAPI;
import com.discord.api.commands.ApplicationCommandType;
import com.discord.models.commands.ApplicationCommandOption;

import java.util.*;

@SuppressWarnings("unused")
public class Gnuify extends Plugin {

    private static final String prefix = "GNU/";

    @NonNull
    @Override
    public Manifest getManifest() {
        Manifest manifest = new Manifest();
        manifest.authors = new Manifest.Author[] { new Manifest.Author("Xinto", 423915768191647755L) };
        manifest.description = String.format("Appends \"%s\" prefix to every word in the sentence.", prefix);
        manifest.version = "1.0.1";
        manifest.updateUrl = "https://raw.githubusercontent.com/X1nto/AliucordPlugins/builds/updater.json";
        return manifest;
    }

    @Override
    public void start(Context context) {
        ApplicationCommandOption argument = new ApplicationCommandOption(ApplicationCommandType.STRING, "sentence", "A sentence to Gnuify", null, true, true, null, null);
        commands.registerCommand(
            "gnuify",
            String.format("Append \"%s\" prefix to every word in the sentence", prefix),
            Collections.singletonList(argument),
            ctx -> {
                String sentence = ctx.getRequiredString("sentence");
                StringBuilder gnuifiedSentence = new StringBuilder();
                for (String word : sentence.split("\\s+")) {
                    gnuifiedSentence.append(prefix).append(word).append(" ");
                }
                return new CommandsAPI.CommandResult(gnuifiedSentence.toString());
            }
        );
    }

    @Override
    public void stop(Context context) {
        commands.unregisterAll();
    }

}

File: plugins/Gnuify/src/main/java/com/aliucord/plugins/Gnuify.kt
package com.aliucord.plugins

import android.content.Context
import com.aliucord.annotations.AliucordPlugin
import com.aliucord.entities.Plugin
import com.aliucord.entities.Plugin.Manifest.Author
import com.aliucord.api.CommandsAPI.CommandResult
import com.discord.api.commands.ApplicationCommandType
import com.discord.models.commands.ApplicationCommandOption
import java.lang.StringBuilder

@AliucordPlugin
class Gnuify : Plugin() {

    private val prefix = "GNU/"

    override fun start(context: Context) {
        val argument = ApplicationCommandOption(
            ApplicationCommandType.STRING,
            "sentence",
            "A sentence to Gnuify",
            null,
            true,
            true,
            null,
            null
        )
        commands.registerCommand(
            "gnuify",
            "Append $prefix prefix to every word in the sentence.",
            listOf(argument)
        ) { ctx ->
            val sentence = ctx.getRequiredString("sentence")
            val gnuifiedSentence = StringBuilder()
            for (word in sentence.split("\\s+").toTypedArray()) {
                gnuifiedSentence.append(prefix).append(word).append(" ")
            }
            return@registerCommand CommandResult(gnuifiedSentence.toString())
        }
    }

    override fun stop(context: Context) {
        commands.unregisterAll()
    }
}